packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3038,10 +3038,11 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// Information set by parent to define where this child fits in its parent's
   /// child list.
   ///
-  /// Subclasses of Element that only have one child should use null for
-  /// the slot for that child.
```

Comment:
Maybe we should remain this info or remove the null check such as `SingleChildRenderObjectElement.insertRenderObjectChild`.

Reply:
as far as i can tell this paragraph was just wrong. most widgets with one child should pass through their own slot to their child, and for the single-child render object widgets it doesn't really matter in most cases but there can definitely be cases where null is the wrong answer. I figured the existing docs in RenderObjectElement were a better source of information.

Reply:
Make sense.